src/ui/public/fancy_forms/kbn_form_controller.js

```diff
@@ -1,74 +1,74 @@
+      super(...superArgs);
 
-  // same as error count, but filters out untouched and pristine models
-  self.softErrorCount = function () {
-    return self.$$invalidModels(function (model) {
-      return model.$touched || model.$dirty;
-    }).length;
-  };
+      $element.on('submit', this._onSubmit);
+      $scope.$on('$destroy', () => {
+        $element.off('submit', this._onSubmit);
+      });
+    }
 
-  self.describeErrors = function () {
-    const count = self.softErrorCount();
-    return count + ' Error' + (count === 1 ? '' : 's');
-  };
+    errorCount() {
+      return this._getInvalidModels().length;
+    }
 
-  self.$$invalidModels = function (predicate) {
-    predicate = _.callback(predicate);
+    // same as error count, but filters out untouched and pristine models
+    softErrorCount() {
+      return this._getInvalidModels()
+        .filter(model => model.$touched || model.$dirty)
+        .length;
+    }
 
-    const invalid = [];
+    describeErrors() {
+      const count = this.softErrorCount();
+      return `${count} Error${count === 1 ? '' : 's'}`;
+    }
 
-    _.forOwn(self.$error, function collect(models) {
-      if (!models) return;
+    $setTouched() {
+      this._getInvalidModels()
+        .forEach(model => model.$setTouched());
+    }
 
-      models.forEach(function (model) {
-        if (model.$$invalidModels) {
-          // recurse into child form
-          _.forOwn(model.$error, collect);
-        } else {
-          if (predicate(model)) {
-            // prevent dups
-            let len = invalid.length;
-            while (len--) if (invalid[len] === model) return;
+    _onSubmit = () => {
+      if (this.errorCount()) {
+        event.preventDefault();
+        event.stopImmediatePropagation();
+        this.$setTouched();
+      }
+    }
 
-            invalid.push(model);
-          }
+    _getInvalidModels() {
+      return this.$$controls.reduce((acc, control) => {
+        // recurse into sub-form
+        if (typeof control._getInvalidModels === 'function') {
+          return [...acc, ...control._getInvalidModels()];
         }
-      });
-    });
-
-    return invalid;
-  };
 
-  self.$setTouched = function () {
-    self.$$invalidModels().forEach(function (model) {
-      // only kbnModels have $setTouched
-      if (model.$setTouched) model.$setTouched();
-    });
-  };
+        if (control.$invalid) {
+          return [...acc, control];
+        }
 
-  function filterSubmits(event) {
-    if (self.errorCount()) {
-      event.preventDefault();
-      event.stopImmediatePropagation();
-      self.$setTouched();
+        return acc;
+      }, []);
     }
   }
 
-  $element.on('submit', filterSubmits);
-  $scope.$on('$destroy', function () {
-    $element.off('submit', filterSubmits);
-  });
+  // replace controller with our wrapper
+  directive.controller = [
+    ...$injector.annotate(KbnFormController),
+    ...$injector.annotate(FormController),
+    (...args) => (
```

Comment:
I agree, but it is necessary to return the `new` result, and writing it as a single line causes the controller call gets a little lost in the noise IMO